### PR TITLE
Add Telegraf Single Chart

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR is the companion to https://github.com/kubernetes/charts/pull/742. Currently the configuration for telegraf is generated by a series of `_helper` functions. There is a PR that will be merged for helm `2.3` that adds a [`toToml`](https://github.com/kubernetes/helm/pull/2050) function that will eliminate the need for this custom code. I will update this PR when that is available in a release.

@jchauncey @jagregory @jeroenvisser101

### CODE REVIEW

**Comment by jackzampolin** [comment]
Renamed and ready for review.

**Comment by jackzampolin** [comment]
Bump?

**Comment by viglesiasce** [comment]
@jackzampolin is this intended to replace the telegraf chart? 

If so you should remove everything but the notes from the original and have the notes point you to installing this one. 

If not, what is the difference?

File: stable/telegraf-standalone/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+version: 0.2.0
+description: Telegraf is an agent written in Go for collecting, processing, aggregating, and writing metrics.
+keywords:
+- telegraf
+- collector
+- timeseries
+- influxdata
+home: https://www.influxdata.com/time-series-platform/telegraf/
+maintainers:
+- name: Jack Zampolin
```
Comment (prydonius) [inline stable/telegraf-standalone/Chart.yaml:11]: Can you change this to `jackzampolin`?

File: stable/telegraf-standalone/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: telegraf-standalone
+version: 0.2.0
+description: Telegraf is an agent written in Go for collecting, processing, aggregating, and writing metrics.
+keywords:
+- telegraf
+- collector
+- timeseries
+- influxdata
+home: https://www.influxdata.com/time-series-platform/telegraf/
```
Comment (prydonius) [inline stable/telegraf-standalone/Chart.yaml:9]: is there an icon for telegraf?

File: stable/telegraf-standalone/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
+# Telegraf
+
+[Telegraf](https://github.com/influxdata/telegraf) is a plugin-driven server agent written by the folks over at [InfluxData](https://influxdata.com) for collecting & reporting metrics.
+
+## TL;DR
+
+```console
+$ helm install stable/telegraf
```
Comment (prydonius) [inline stable/telegraf-standalone/README.md:8]: looks like this readme needs to be updated to reference `telegraf-standalone`

File: stable/telegraf-standalone/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+- kubectl exec -i -t --namespace {{ .Release.Namespace }} $(kubectl get pods --namespace {{ .Release.Namespace }} -l app={{ template "fullname" . }} -o jsonpath='{.items[0].metadata.name}') /bin/sh
+
+To tail the logs for a Telegraf pod run the following:
+
+- kubectl logs -f --namespace {{ .Release.Namespace }} $(kubectl get pods --namespace {{ .Release.Namespace }} -l app={{ template "fullname" . }} -o jsonpath='{ .items[0].metadata.name }')
+
+{{- if empty .Values.config.outputs.influxdb.urls }}
+
+Need to set an InfluxDB url for either s daemonset to deploy this instance:
```
Comment (prydonius) [inline stable/telegraf-standalone/templates/NOTES.txt:11]: > either s daemonset

typo?

File: stable/telegraf-standalone/templates/_helpers.tpl [hunk 5]
```diff
@@ -0,0 +1,164 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 24 -}}
```
Comment (prydonius) [inline stable/telegraf-standalone/templates/_helpers.tpl:6]: Please update this to `trunc 63 | trimSuffix "-"`

File: stable/telegraf-standalone/templates/_helpers.tpl [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 24 -}}
```
Comment (prydonius) [inline stable/telegraf-standalone/templates/_helpers.tpl:15]: same as above

**Review by prydonius — COMMENTED** [review]
@viglesiasce @jackzampolin should we do the deprecation of `telegraf` in a separate PR?

**Comment by unguiculus** [comment]
What's the status here? Please update within next week.

**Comment by viglesiasce** [comment]
Closing as stale

**Pull request closed without merging**